docs/proposals/formatted-output.md

```diff
@@ -0,0 +1,359 @@
+# Formatted ORAS CLI output
+
+## Table of Contents
+
+- [Background](#background)
+- [Guidance](#guidance)
+- [Scenarios](#scenarios)
+  - [Scripting](#scripting)
+  - [CI/CD](#cicd)
+- [Proposal and desired user experience](#proposal-and-desired-user-experience)
+  - [oras pull](#oras-pull)
+  - [oras push](#oras-push)
+  - [oras attach](#oras-attach)
+  - [oras discover](#oras-discover)
+- [FAQ](#faq)
+
+## Background
+
+ORAS has prettified output designed for humans. However, for machine processing, especially in automation scenarios, like scripting and CI/CD pipelines, developers want to perform batch operations and chain different commands with ORAS, as well as filtering, modifying, and sorting objects based on the outputs that are emitted by the ORAS command. Developers expect that ORAS output can be emitted as machine-readable text instead of only the prettified or tabular data, so that it can be used to perform other operations.
+
+The formatted output is not intended to supersede the prettified human-readable and friendly output text of ORAS CLI. It aims to provide a programming-friendly experience for developers who want to automate their workflows without needing to parse unstructured text.
+
+## Guidance
+
+Provide two major options to enable users to define the output format of ORAS commands:
+
+- Use `--output` to output a file or directory in the filesystem
+- Use `--format` to format the output of ORAS commands into different data formats or enable process output data using the given Go template
+  - Use `--format json|tree|table` to print the output in prettified JSON, tree view, or table view
+  - Use `--format '{{ GO_TEMPLATE_FUNCTION }}'` to enable extract or compose the output data using Go template functions 
+
+For example, when using `oras manifest fetch` with the flag `--format`, the following fields should be formatted into JSON output:
```

Comment:
Should the `oras manifest fetch` part be moved to the section `Proposal and desired user experience`?

Reply:
I will move it to the section `Proposal and desired user experience` below.